core/domain/base_model_validators.py

```diff
@@ -84,12 +89,75 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
-        """
+
+        Raises:
+            Exception. This class was used instead of
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
+        """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'When fetching instances of UserSettingsModel, please use ' +
+                'UserSettingsModelFetcherDetails instead of ' +
+                'ExternalModelFetcherDetails')
         self.field_name = field_name
         self.model_class = model_class
         self.model_ids = model_ids
 
 
+class UserSettingsModelFetcherDetails(python_utils.OBJECT):
+    """Value object providing ids to fetch the user settings model."""
+
+    def __init__(
+            self, field_name, model_ids,
+            may_contain_system_ids, may_contain_pseudonymous_ids):
+        """Initializes the UserSettingsModelFetcherDetails domain object.
+
+        Args:
+            field_name: str. A specific name used as an identifier by the
+                storage model which is used to identify the user settings model
+                reference. For example: 'committer_id': UserSettingsModel,
+                committer_id is the field name to identify the external model
+                UserSettingsModel.
+            model_ids: list(str). The list of user settings model IDs to fetch
+                the UserSettingsModels.
+            may_contain_system_ids: bool. Whether the model IDs contain
+                system IDs which should be omitted before attempting to fetch
+                the corresponding models. Set may_contain_system_ids to True if
+                and only if this field can contain admin or bot IDs.
+            may_contain_pseudonymous_ids: bool. Whether the model ids contain
+                pseudonymous IDs which should be omitted before attempting to
+                fetch the corresponding models. Set may_contain_pseudonymous_ids
+                to True if and only if this field can contain user IDs that
+                are pseudonymized as part of Wipeout. These fields can only be
```

Comment:
These fields --> In other words, these fields

Reply:
Done.